toolchain/sem_ir/formatter.cpp

```diff
@@ -952,10 +987,26 @@ class Formatter {
   bool pending_constant_value_is_self_ = false;
 };
 
-auto FormatFile(const Lex::TokenizedBuffer& tokenized_buffer,
-                const Parse::Tree& parse_tree, const File& sem_ir,
-                llvm::raw_ostream& out) -> void {
-  Formatter(tokenized_buffer, parse_tree, sem_ir, out).Format();
+Formatter::Formatter(const Lex::TokenizedBuffer& tokenized_buffer,
+                     const Parse::Tree& parse_tree, const File& sem_ir)
+    : formatter_(std::make_unique<FormatterImpl>(tokenized_buffer, parse_tree,
+                                                 sem_ir)) {}
+
+Formatter::~Formatter() = default;
+
+auto Formatter::Print(llvm::raw_ostream& out) -> void {
+  formatter_->Print(out);
+}
+
+auto Formatter::PrintPartialTrailingCodeBlock(
+    llvm::ArrayRef<SemIR::InstId> block, int indent, llvm::raw_ostream& out)
+    -> void {
+  formatter_->PrintPartialTrailingCodeBlock(block, indent, out);
+}
+
+auto Formatter::PrintInst(SemIR::InstId inst_id, int indent,
+                          llvm::raw_ostream& out) -> void {
+  formatter_->PrintInst(inst_id, indent, out);
 }
```

Comment:
Instead of the pimpl approach and saving/restoring members, perhaps we could create an instance of `FormatterImpl` in each of these `Formatter::Print` functions? We don't actually want to persist any state between them other than the `InstNamer` (which the `Formatter` could still own).

Reply:
Done